### ISSUE TITLE
(Feature request) render clipped html pages more pretty

### ISSUE THREAD

**Issue by ZhengfengRao** [zadam/trilium#4243]
### Describe feature

Web pages clipped in trilium, rendered ugly:
<img width="1199" alt="image" src="https://github.com/zadam/trilium/assets/1232937/0f2a93f1-8f8b-418b-8f37-cc94da39eb57">

clipped by singlefile, and rendered by Chrome:
<img width="1012" alt="image" src="https://github.com/zadam/trilium/assets/1232937/80a9dfba-97e5-4936-918f-2efc522a0e5b">

can we render the html pages as Chrome?

### Additional Information

palftorm: mac
version: 版本0.60.4 (0.60.4)

**Comment by zadam** [comment]
No, it's not possible. Rendering HTML pages exactly as in chrome would mean storing all the pages assets like CSS, images, fonts, JavaScripts and it's just too much. A page could take many megabytes.

Instead, what we do is that we run "readability" script which simplifies the page and tries to store the semantic content.

A workaround if you want exact look, store a screenshot of the page.

**Comment by Nriver** [comment]
You can try https://github.com/nil0x42/singlefile2trilium, if you want to capture a webpage exactly as it appears in your web browser.

**Comment by ZhengfengRao** [comment]
> No, it's not possible. Rendering HTML pages exactly as in chrome would mean storing all the pages assets like CSS, images, fonts, JavaScripts and it's just too much. A page could take many megabytes.
> 
> Instead, what we do is that we run "readability" script which simplifies the page and tries to store the semantic content.
> 
> A workaround if you want exact look, store a screenshot of the page.

thanks for your reply.

The singleFile plugin stores all of the resources(css, js, images) in a single html file, all we need to do is render it properly, maybe we can choose other webview to render?(i'm not an expert on web and electron, i don't know whether its possible)

**Comment by ZhengfengRao** [comment]
> You can try https://github.com/nil0x42/singlefile2trilium, if you want to capture a webpage exactly as it appears in your web browser.
> 
> The drawback for this method is that the content is not editable or searchable.

thanks.

i've tried this plugin, and even import html pages saved by singleFile plugin manually, however pages rendered as i'd posted above.

**Comment by zadam** [comment]
The plugin is possibly no longer compatible with newer versions of Trilium.

**Comment by Nriver** [comment]
This is my slightly modified version of the plugin which is included in the demo.zip of Chinese release. It's working in 0.60.4

https://github.com/Nriver/trilium-translation/tree/main/demo-cn/示例笔记%20-%20请不要删除/Trilium%20扩展/Trilium%20自定义API/网页快照%20singlefile2trilium

![ksnip_20230914-093903](https://github.com/zadam/trilium/assets/6752679/9cc12480-447e-4fe4-8a03-fcacd37528d0)

**Comment by ZhengfengRao** [comment]
> This is my slightly modified version of the plugin which is included in the demo.zip of Chinese release. It's working in 0.60.4
> 
> [https://github.com/Nriver/trilium-translation/tree/main/demo-cn/示例笔记%20-%20请不要删除/Trilium%20扩展/Trilium%20自定义API/网页快照%20singlefile2trilium](https://github.com/Nriver/trilium-translation/tree/main/demo-cn/%E7%A4%BA%E4%BE%8B%E7%AC%94%E8%AE%B0%20-%20%E8%AF%B7%E4%B8%8D%E8%A6%81%E5%88%A0%E9%99%A4/Trilium%20%E6%89%A9%E5%B1%95/Trilium%20%E8%87%AA%E5%AE%9A%E4%B9%89API/%E7%BD%91%E9%A1%B5%E5%BF%AB%E7%85%A7%20singlefile2trilium)
> 
> ![ksnip_20230914-093903](https://user-images.githubusercontent.com/6752679/267827341-9cc12480-447e-4fe4-8a03-fcacd37528d0.png)

it looks great!
I'll try it,thanks for your nice work.

**Comment by ZhengfengRao** [comment]
> This is my slightly modified version of the plugin which is included in the demo.zip of Chinese release. It's working in 0.60.4
> 
> [https://github.com/Nriver/trilium-translation/tree/main/demo-cn/示例笔记%20-%20请不要删除/Trilium%20扩展/Trilium%20自定义API/网页快照%20singlefile2trilium](https://github.com/Nriver/trilium-translation/tree/main/demo-cn/%E7%A4%BA%E4%BE%8B%E7%AC%94%E8%AE%B0%20-%20%E8%AF%B7%E4%B8%8D%E8%A6%81%E5%88%A0%E9%99%A4/Trilium%20%E6%89%A9%E5%B1%95/Trilium%20%E8%87%AA%E5%AE%9A%E4%B9%89API/%E7%BD%91%E9%A1%B5%E5%BF%AB%E7%85%A7%20singlefile2trilium)
> 
> ![ksnip_20230914-093903](https://user-images.githubusercontent.com/6752679/267827341-9cc12480-447e-4fe4-8a03-fcacd37528d0.png)

feedback:
i've tried singlefile2trilium, it seems to be some issue in the origin code on Mac.

the following is web page clipped by origin code of singlefile2trilium, it can't be rendered on my Mac:
![image](https://github.com/zadam/trilium/assets/1232937/d116985c-a88a-481a-b4dd-b54e71c7ad75)

i exported the note as zipped html, and open in chrome, it reported errors:
<img width="1436" alt="image" src="https://github.com/zadam/trilium/assets/1232937/fbc6b4b1-11a5-4ca0-a544-d473a93d6338">
it seems to be a issue of iframe, i guess it my be security issue of Chrome on Mac, and also Trilum.

but the exported page can be displayed on Safari perfectly.

at last, i modified the code of singlefile2trilium-handler.js, by removing the wrapped "r-iframe" as flowing:
![image](https://github.com/zadam/trilium/assets/1232937/3b07059a-f631-42b3-a70f-1796455930a1)

it seems to be good this time:
![image](https://github.com/zadam/trilium/assets/1232937/fa4450a5-c80c-4d89-b352-372521f2fd59)

in short, i create note with the origin bytes  saved by singlefile, don't add wrapped iframe.
i don't know whether this workaround will cause any side effect.

**Comment by rauenzi** [comment]
Using the SingleFile source directly can definitely have side effects on the application. When used like that any conflicting styles or scripts will affect Trilium

**Comment by ZhengfengRao** [comment]
> Using the SingleFile source directly can definitely have side effects on the application. When used like that any conflicting styles or scripts will affect Trilium

exactly.
some clipped pages float on title bar of trilium.
![image](https://github.com/zadam/trilium/assets/1232937/3405364d-8db4-4234-b707-ed2086506a99)

**Comment by rauenzi** [comment]
I actually just released an addon that imports SingleFile saved pages if you'd be interested in using that. There's a preview and download/install instructions on the repo https://github.com/rauenzi/Trilium-SingleFile It uses an iframe to prevent issues like that but still keeps the original source intact.

**Comment by ZhengfengRao** [comment]
> I actually just released an addon that imports SingleFile saved pages if you'd be interested in using that. There's a preview and download/install instructions on the repo https://github.com/rauenzi/Trilium-SingleFile It uses an iframe to prevent issues like that but still keeps the original source intact.

i've tried your plugin, it renders perfectly on my mac, really a nice work! absolutely the one i'v been expecting.
however, can the page be rendered in full content of note? in another word, can we render the page in the green box?
<img width="1251" alt="image" src="https://github.com/zadam/trilium/assets/1232937/5f2baef1-5046-4bd1-baec-c9e0d144221e">

**Comment by rauenzi** [comment]
@ZhengfengRao I saw your comment before the edit, if you'd like help making it full page create an issue on the Trilium-SingleFile repo and I'll help you out

**Comment by ZhengfengRao** [comment]
> @ZhengfengRao I saw your comment before the edit, if you'd like help making it full page create an issue on the Trilium-SingleFile repo and I'll help you out

i modified the css, and will submit a PR later
